pkg/network/tracer/tracer_windows.go

```diff
@@ -301,3 +331,28 @@ func newUSMMonitor(c *config.Config, dh driver.Handle) usm.Monitor {
 	monitor.Start()
 	return monitor
 }
+
+func (t *Tracer) addProcessInfo(c *network.ConnectionStats) {
+	if t.processCache == nil {
+		return
+	}
+
+	c.ContainerID.Source, c.ContainerID.Dest = nil, nil
+
+	// on windows, cLastUpdateEpoch is already set as
+	// ns since unix epoch.
+	ts := c.LastUpdateEpoch
+	p, ok := t.processCache.Get(c.Pid, int64(ts))
+	if !ok {
+		return
+	}
+
+	if len(p.Tags) > 0 {
+		c.Tags = make([]*intern.Value, len(p.Tags))
+		copy(c.Tags, p.Tags)
+	}
+
+	if p.ContainerID != nil {
```

Comment:
Have we tested this when Windows containers are running on the host? Afaik, we don't support containers at all for USM, regardless of if they're running on the host where the driver is installed, but I don't know if the driver or Agent makes any attempt to distinguish between processes running in a container or directly on the host, so this could give unexpected results if we happen to be running in the presence of containers.

So, I 'm not really sure what the best path forward is here - what do you think?

Reply:
I have not (yet) tried in the presence of containers.

Reply:
FYI one of the Windows USM pre-reqs is: "Your service is running on a virtual machine."